document-generator-accounts/src/main/java/uk/gov/companieshouse/document/generator/accounts/AccountsDocumentInfoServiceImpl.java

```diff
@@ -26,25 +27,25 @@
     private static final Logger LOG = LoggerFactory.getLogger(MODULE_NAME_SPACE);
 
     @Override
-    public DocumentInfo getDocumentInfo() {
+    public DocumentInfoResponse getDocumentInfo(DocumentInfoRequest documentInfoRequest) {
         LOG.info("Started getting document");
 
-        String resource = "";
-        String resourceId = "";
+        String resourceId = documentInfoRequest.getResourceId();
+        String resourceUri = documentInfoRequest.getResourceUri();
 
-        Transaction transaction = transactionService.getTransaction(resource);
+        Transaction transaction = transactionService.getTransaction(resourceId);
         if (transaction == null) {
             LOG.error("transaction not found");
             return null;
         }
 
         String resourceLink =  Optional.of(transaction)
                 .map(Transaction::getResources)
-                .map(resources -> resources.get(resourceId))
+                .map(resources -> resources.get(resourceUri))
```

Comment:
This should be the resourceId. It's the ID of the resource within the transaction.